widgets/matplotlib/simulation/recoil_atom_distribution.py

```diff
@@ -799,6 +800,37 @@ def __create_percent_widget(self):
     def open_recoil_element_info(self):
         """Open recoil element info.
         """
+        dialog = ReferenceDensityDialog(
+            self.current_recoil_element, self.current_element_simulation)
+        if dialog.isOk:
+            new_values = dialog.get_properties()
+            try:
+                old_recoil_name = self.current_recoil_element.name
+                self.current_element_simulation.update_recoil_element(
+                    self.current_recoil_element,
+                    new_values)
+                # If name has changed
+                if old_recoil_name != self.current_recoil_element.name:
+                    # Delete energy spectra that use recoil
+                    df.delete_recoil_espe(
+                        self.tab,
+                        f"{self.current_recoil_element.prefix}-"
+                        f"{old_recoil_name}")
+
+                    self.recoil_name_changed.emit(
+                        self.current_element_simulation,
+                        self.current_recoil_element)
+                self.update_recoil_element_info_labels()
+            except KeyError:
+                error_box = QtWidgets.QMessageBox()
+                error_box.setIcon(QtWidgets.QMessageBox.Warning)
+                error_box.addButton(QtWidgets.QMessageBox.Ok)
+                error_box.setText(
+                    "All recoil element information could not be saved.")
+                error_box.setWindowTitle("Error")
+                error_box.exec()
```

Comment:
`QMessageBox.critical()` would simplifiy this. Search the code for examples.

Also, the error message should probably be visible to the user.

Reply:
I haven't done this but I'll check it.

Reply:
Check